qiskit/providers/aer/openpulse/duffing_model_generators.py

```diff
@@ -0,0 +1,438 @@
+                         oscillator_freqs,
+                         anharm_freqs,
+                         drive_strengths,
+                         coupling_dict,
+                         dt,
+                         freq_symbol='v',
+                         anharm_symbol='alpha',
+                         drive_symbol='r',
+                         coupling_symbol='j'):
+    """Returns a PulseSystemModel for a specified duffing oscillator system, and a dict specifying
+    ControlChannel indices for cross-resonance driving.
+
+    Single oscillators are specified by three parameters: frequency v, anharmonicity alpha, and
+    drive strength r, which enter into the Hamiltonian model via the terms:
+        pi*(2*v-alpha)*O + pi*alpha*O*O + 2*pi*r*X*D(t),
+    where O is the number operator, X is the drive operator, and D(t) is the signal of the
+    corresponding DriveChannel.
+
+    Couplings between oscillators are specified in the argument coupling_dict, with keys being
+    edges, and values being the coupling strenghts. Couplings enter the Hamiltonian model via an
+    exchange coupling term:
+        2*pi*j*(A0*C1+C0*A1),
+    where j is the coupling strength, and A0*C1+C0*A1 is the exchange coupling operator between the
+    two oscillators, where A is the annihiliation operator, and C is the creation operator.
+
+    For each pair of coupled oscillators, pulse ControlChannels are prepared for doing cross
+    resonance (CR) drives between oscillators. A CR drive on oscillators 0 with target oscillators
+    1 is modeled with the Hamiltonian term:
+        2*pi*r0*X0*U(t),
+    where r0 is the drive strength for oscillators 0, X0 is the drive operator for oscillators 0,
+    and U(t) is the signal for the CR drive. U(t) is set to have carrier frequency equal to that of
+    the target oscillators.
+
+    Indices for ControlChannels corresponding to CR drives are provided in the returned cr_idx_dict.
+
+    Args:
+        num_oscillators (int): Number of oscillators in the model
+        dim_oscillators (int): Dimension of truncation for each oscillator
+        oscillator_freqs (list): Oscillator frequencies, assumed to be of length num_oscillators
+        anharm_freqs (list): Anharmonicity values, assumed to be of length num_oscillators
+        drive_strengths (list): Drive strength values, assumed to be of length num_oscillators
+        coupling_dict (dict): Specification of the coupling graph with keys being edges, and values
+                              the coupling strengths.
+
+                              For example:
+
+                              * ``{(0,1): 0.02, (1,3): 0.01}`` specifies a system in which
+                                oscillators (0,1) are coupled with strength 0.02, and (1,3) are
+                                coupled with strength 0.01
+        dt (float): Pixel size for pulse instructions
+        freq_symbol (str): Frequency symbol for internal HamiltonianModel
```

Comment:
What is the purpose of these symbol kwargs? Will a user ever need to override them or can we just set them to their default values inside the function?

Reply:
I put them there due to the fact that, unfortunately, the symbols survive in the background of `HamiltonianModel`, and so it could be useful to have some control over this. You're right though that at the moment there is no use for this, so I can remove them.